flink-cyber/metron-parser-chain/parser-chains-core/src/main/java/com/cloudera/parserchains/core/ReflectiveParserBuilder.java

```diff
@@ -68,19 +70,53 @@ private void configure(Parser parser, ParserSchema parserSchema) throws InvalidP
 
             final List<ConfigValueSchema> valuesSchema = parserConfig.get(annotationKey);
 
+            validateParams(parserSchema, valuesSchema, method);
+
             if (valuesSchema != null) {
```

Comment:
this if structure looks a bit messy especially with validation checking and default value assignment logic.